src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/storm/bolt/isl/IslHandler.java

```diff
@@ -219,17 +220,18 @@ public void islRemovedNotification(Endpoint srcEndpoint, IslReference reference)
     public void islChangedNotifyFlowMonitor(IslReference reference) {
         Optional<Endpoint> src = Optional.ofNullable(reference.getSource());
         Optional<Endpoint> dst = Optional.ofNullable(reference.getDest());
-        IslChangedInfoData islChangedInfoData = IslChangedInfoData.builder()
+        IslChangedInfoDataBuilder islChangedInfoData = IslChangedInfoData.builder()
                 .source(NetworkEndpoint.builder()
                         .datapath(src.map(Endpoint::getDatapath).orElse(null))
                         .portNumber(src.map(Endpoint::getPortNumber).orElse(null))
-                        .build())
-                .destination(NetworkEndpoint.builder()
-                        .datapath(dst.map(Endpoint::getDatapath).orElse(null))
-                        .portNumber(dst.map(Endpoint::getPortNumber).orElse(null))
-                        .build())
-                .build();
-        emit(STREAM_FLOW_MONITORING_ID, getCurrentTuple(), makeIslChangedTuple(islChangedInfoData));
+                        .build());
+        if (dst.isPresent()) {
```

Comment:
How it can be? Isl can't operate on incomplete IslReference, so it can't (must not) appear here.

Reply:
It's special case when IslReference with only one side created to notify flow monitoring about isl delete.

Reply:
Why not to fill both ends?

Reply:
So you make a special treatment of `IslChangedInfoData` with only source part filled? As for me must be used new / separate event, to notify about ISL removal instead of such weird unexpected logic built in into existing object even not into object, but in some handler processing this object.

Reply:
I used this way to use only one message type (`IslChangedInfoData`) to flow-monitoring topology about any ISL changes (moved or deleted). I will try to rework it using separate message type.